plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryCaseInsensitiveMapping.java

```diff
@@ -39,11 +41,22 @@
 {
     private BigQuerySqlExecutor bigQuerySqlExecutor;
 
+    @BeforeClass
+    public void initBigQueryExecutor()
+    {
+        this.bigQuerySqlExecutor = new BigQuerySqlExecutor();
```

Comment:
Why not make the field protected if you want to decouple the creation of `bigQuerySqlExecutor` from init? Any subclass/test which wants to not/differently create `bigQuerySqlExecutor` can then override `createQueryRunner`.

This is exactly what we do in other connectors - e.g. https://github.com/trinodb/trino/blob/2c257df99b586422114421ad08959eebb0c1c481/plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlConnectorTest.java#L69

Reply:
Ok we can make this field protected and extract another method. 

the method itself is called createQueryRunner() and that's the only thing that it should do. If there is a need to create something else, like BigQuerySqlExecutor in this example, it should be done by a separate setup method. Thanks to that we don't mix up the responsibility and when you want to create a QueryRunner in a different way, you don't need to create another object that is in no direct way connected to a QueryRunner  do so. It's just a bad design and I want to fix it when it's easy